wagtail/admin/templates/wagtailadmin/pages/confirm_delete.html

```diff
@@ -68,6 +68,19 @@
 
         <form action="{% url 'wagtailadmin_pages:delete' page.id %}" method="POST">
             {% csrf_token %}
+            {% if confirm_before_delete %}
+                <p class="status-msg warning">
+                    {% blocktrans trimmed with total_pages=descendant_count|add:1 %}
+                        This action will delete total <b>{{ total_pages }}</b> pages.
+                    {% endblocktrans %}
+                </p>
+                <label class="no-float w-font-normal" for="id_confirm_site_name">
+                    {% blocktrans trimmed %}
+                        Please type <b>{{ wagtail_site_name }}</b> to confirm.
+                    {% endblocktrans %}
+                </label>
+                <input class="w-mb-4" type="text" name="confirm_site_name" id="id_confirm_site_name" required>
```

Comment:
Bonus point: make the warning message associated with the input for screen reader users.

```suggestion
                <input class="w-mb-4" type="text" name="confirm_site_name" id="id_confirm_site_name" aria-describedby="id_confirm_site_name-warning" required>
```